sdk/couchbase-core/src/memdx/error.rs

```diff
@@ -206,8 +206,12 @@ pub enum ServerErrorKind {
     KeyNotFound,
     #[error("Temporary failure")]
     TmpFail,
+    #[error("CAS mismatch")]
+    CasMismatch,
```

Comment:
Should `CasMismatch` be a `ServerErrorKind`, or something different?

Reply:
Seems like a `ServerErrorKind` to me. The only place it can originate is from the server.